### PR TITLE
Move deploy form outside the drawer to a page

### PR DESCRIPTION
**Pull request by MananTank**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
The focus of this PR is to enhance the contract publishing process by adding deploy functionality and improving UI components.

### Detailed summary
- Added deploy functionality to the contract publishing page
- Updated the `NetworkSelectorButton` component with a new theme method
- Modified the `ShareButton` component to include current URL
- Improved the UI components for better user experience in publishing pages

> The following files were skipped due to too many changes: `apps/dashboard/src/components/pages/publish.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by MananTank** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3936?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3936** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3936?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3928** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3928?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3926** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3926?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @MananTank and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/components/share-buttom/index.tsx [hunk 1]
```diff
@@ -2,9 +2,10 @@ import { Icon, useClipboard, useToast } from "@chakra-ui/react";
...
-export const ShareButton: React.FC<Required<Omit<ShareData, "files">>> = (
-  shareData,
-) => {
+export const ShareButton: React.FC<ShareData> = (props) => {
+  const currentUrl = typeof window !== "undefined" ? window.location.href : "";
+  const url = props.url || currentUrl;
+  const shareData = { ...props, url };
   const { onCopy, hasCopied } = useClipboard(shareData.url);
 
   const toast = useToast();
```
Comment (jnsdls) [inline apps/dashboard/src/components/share-buttom/index.tsx:11]: might as well move this to `sonner` now, right?

**Review by jnsdls — APPROVED** [review]
looks like everything is working

File: apps/dashboard/src/components/share-buttom/index.tsx (hunk 1 above)
Comment (MananTank) [reply apps/dashboard/src/components/share-buttom/index.tsx:11]: I'll change this later iterations

**Pull request merged**